controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -276,15 +276,15 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 
 		}
 
-		if availableServices.Mcg {
+		if availableServices.Mcg && r.storageConsumer.Status.Client != nil {
 			// A provider cluster already has a NooBaa system and does not require a NooBaa account
 			// to connect to a remote cluster, unlike client clusters.
 			// A NooBaa account only needs to be created if the storage consumer is for a client cluster.
 			clusterID := util.GetClusterID(r.ctx, r.Client, &r.Log)
-			clientStatus := &r.storageConsumer.Status.Client
+			clientClusterID := r.storageConsumer.Status.Client.ClusterID
```

Comment:
This makes assumptions that we have client on the status. that might not be the case.